onnxruntime/core/util/thread_utils.cc

```diff
@@ -10,40 +10,65 @@
 #endif
 #include <thread>
 #include "core/session/ort_apis.h"
+#include "core/common/string_utils.h"
+#include "core/common/logging/logging.h"
 
 namespace onnxruntime {
 namespace concurrency {
+
+// extract affinity from affinity string
+// processor id in affinity string starts from 1
+std::vector<LogicalProcessors> ReadThreadAffinityConfig(const std::string& affinity_str) {
+  std::vector<LogicalProcessors> logical_processors_vector;
+  auto affinities = utils::SplitString(affinity_str, ";");
+  for (const auto& affinity : affinities) {
+    LogicalProcessors logical_processors;
+    auto processor_interval = utils::SplitString(affinity, "-");
+    if (processor_interval.size() == 2) {
+      auto processor_from = std::stoi(processor_interval[0].data());
+      auto processor_to = std::stoi(processor_interval[1].data());
+      ORT_ENFORCE(processor_from > 0 && processor_to > 0,
+                  std::string{"Processor id must starts from 1: "} + affinity.data());
+      ORT_ENFORCE(processor_from <= processor_to,
+                  std::string{"Invalid processor interval: "} + affinity.data());
+      logical_processors.resize(static_cast<size_t>(1ULL + processor_to - processor_from));
```

Comment:
>ogical_processors.resize(static_cast<size_t>(1ULL + processor_to - processor_from)); [](http://example.com/codeflow?start=7&length=84)

What is the expected result of this resize()?
For a range, as I understand, we need to feel consecutive numbers as logical processor Ids. However, resize() initializes all entries to a single value.